### PR TITLE
Refactoring code to move the code in appropriate folders

### PR DESCRIPTION
**Pull request by richajaindce**

### CODE REVIEW

**Comment by richajaindce** [comment]
- Lets also move field.rs to some folder - any suggestions where?
- I wasn't able to move net/ due to compile error due to an error - DeadThread which should ideally sit inside ipa_bench @taikiy can you help on this

File: src/secret_sharing_schemes/mod.rs [hunk 1]
```diff
@@ -0,0 +1,2 @@
+pub mod replicated_secret_sharing;
```
Comment (akoshelev) [inline src/secret_sharing_schemes/mod.rs:1]: I don't think we need to make the whole module visible, just re-exporting secret sharing structs is enough

File: src/protocol/securemul.rs [hunk 2]
```diff
@@ -247,19 +226,19 @@ mod tests {
     use rand::Rng;
     use rand_core::RngCore;
 
-    use crate::replicated_secret_sharing::ReplicatedSecretSharing;
+    use crate::secret_sharing_schemes::replicated_secret_sharing::ReplicatedSecretSharing;
```
Comment (akoshelev) [inline src/protocol/securemul.rs:229]: secret sharing is repeated 3 times here, maybe we can have `secret_sharing` module that exports `Replicated` and `Shamir`?

File: src/protocol/mod.rs [hunk 3]
```diff
@@ -1,3 +1,6 @@
+pub mod context;
+pub mod securemul;
```
Comment (akoshelev) [inline src/protocol/mod.rs:2]: ditto: lets just export `SecureMul` struct so code can import it as `use crate::protocol::SecureMul`

**Comment by akoshelev** [comment]
`chunkscan` module does not technically belong to `helpers` as it is just a generic stream ext, so maybe we can keep it in the root folder for now

**Review by akoshelev — COMMENTED** [review]
Thanks for making this change @richajaindce. It's been bugging me for a while too and I am glad that you took the lead here

**Comment by richajaindce** [comment]
> `chunkscan` module does not technically belong to `helpers` as it is just a generic stream ext, so maybe we can keep it in the root folder for now
I was struggling to put this as well as field.rs in a module, how about we move both to another folder named common?

**Comment by akoshelev** [comment]
> > `chunkscan` module does not technically belong to `helpers` as it is just a generic stream ext, so maybe we can keep it in the root folder for now
> > I was struggling to put this as well as field.rs in a module, how about we move both to another folder named common? @akoshelev

I think it is fine if we cannot find a new home for every file in the root folder. there is a high chance that `chunkscan` will go away and be deleted very soon and `field` currently is just one implementation so imo it is fine to keep them where they are

File: Cargo.toml [hunk 4]
```diff
@@ -6,6 +6,7 @@ edition = "2021"
 
 [features]
 default = ["debug", "cli"]
+bin = []
```
Comment (akoshelev) [inline Cargo.toml:9]: don't see this feature being used inside the code. am I missing it?

File: src/bin/ipa_bench/net/client/mod.rs [hunk 5]
```diff
@@ -6,6 +6,7 @@ use thiserror::Error as ThisError;
 
 use super::Command;
 
+#[allow(dead_code)]
```
Comment (akoshelev) [inline src/bin/ipa_bench/net/client/mod.rs:9]: it is not ideal to sprinkle these suppressions everywhere, in my experience it usually happens when I forget to declare module public. is it the same here?

Comment (richajaindce) [reply src/bin/ipa_bench/net/client/mod.rs:9]: @taikiy would you intend to use this code in MpcHandle and MpcHttpConnection or is it safe to purge that code for now?

File: Cargo.toml (hunk 4 above)
Comment (richajaindce) [reply Cargo.toml:9]: some of the constructs in ipa_bench folder i.e test_mpc, helper and ipa_bench are using it. CC: @taikiy

Comment (akoshelev) [reply Cargo.toml:9]: can you folks point me to one of those constructs? I don't seem to be able to find it

**Review by akoshelev — APPROVED** [review]
Thanks for cleaning this up!

**Pull request merged**